packages/app-store/salesforce/lib/CrmService.ts

```diff
@@ -329,9 +329,26 @@ export default class SalesforceCRMService implements CRM {
     }
     const results = await conn.query(soql);
 
-    if (!results || !results.records.length) return [];
+    let records: ContactRecord[];
+
+    // If falling back to contacts, check for the contact before returning the leads or empty array
+    if (
+      appOptions.createEventOn === SalesforceRecordEnum.LEAD &&
+      appOptions.createEventOnLeadCheckForContact
+    ) {
+      // Get any matching contacts
+      const contactSearch = await conn.query(
+        `SELECT Id, Email, OwnerId FROM ${SalesforceRecordEnum.CONTACT} WHERE Email IN ('${emailArray.join(
+          "','"
+        )}')`
+      );
+
+      if (contactSearch || contactSearch.records.length) records = contactSearch.records as ContactRecord[];
```

Comment:
```suggestion
      if (contactSearch.records?.length > 0) records = contactSearch.records as ContactRecord[];
```

shouldn't we use this instead?